### PR TITLE
Fix camera following tux when dying 

### LINKED ISSUE #2799: Camera keeps following Tux when dying
As of recently when Tux dies, the Camera keeps following him rather than remaing stationary.
Most noticeable in tall levels and dying high up.

### PR DESCRIPTION
**Pull request by Miner34dev**
fixes #2799

### CODE REVIEW

File: src/object/camera.cpp [hunk 1]
```diff
@@ -310,31 +310,34 @@ Camera::draw(DrawingContext& context)
...
-    default:
-      break;
+  if (Sector::current() && !(Sector::current()->get_players()[0]->is_dying()))
+  {
+    // Minimum scale should be set during the update sequence; else, reset it.
+    m_enfore_minimum_scale = false;
+
+    switch (m_mode) {
+      case Mode::NORMAL:
+        if (Sector::current() && Sector::current()->get_object_count<Player>() > 1)
```
Comment (tobbi) [inline src/object/camera.cpp:320]: Redunant check for `Sector::current()`

**Comment by Vankata453** [comment]
Perhaps this check should only be done right before `update_scroll_normal()`? This will allow for it to not include multiplayer (even so, it currently only checks for the first player anyway), as well as any other custom modes.

**Comment by Miner34dev** [comment]
Anything else to change?  I'm ready to squash the commits

**Comment by Vankata453** [comment]
> Perhaps this check should only be done right before `update_scroll_normal()`? This will allow for it to not include multiplayer (even so, it currently only checks for the first player anyway), as well as any other custom modes.

What about this?

File: src/object/camera.cpp [hunk 2]
```diff
@@ -319,7 +319,7 @@ Camera::update(float dt_sec)
       {
         update_scroll_normal_multiplayer(dt_sec);
       }
-      else
+      else if (Sector::current() && !(Sector::current()->get_players()[0]->is_dying()))
```
Comment (Vankata453) [inline src/object/camera.cpp:322]: Seeing the `if` above this also checks for `Sector::current()`, both statements could be nested in an `if (Sector::current())` to not check twice.

**Review by Vankata453 — APPROVED** [review]

**Review by mrkubax10 — APPROVED** [review]

**Review by tobbi — APPROVED** [review]

**Pull request merged**